weed/s3api/s3api_object_handlers.go

```diff
@@ -376,6 +376,13 @@ func (s3a *S3ApiServer) proxyToFiler(w http.ResponseWriter, r *http.Request, des
 	s3a.maybeAddFilerJwtAuthorization(proxyReq, isWrite)
 	resp, postErr := s3a.client.Do(proxyReq)
 
+	if resp.Uncompressed && r.Header.Get("Accept-Encoding") == "" {
+		r.Header.Set("Accept-Encoding","gzip")
+		util.CloseResponse(resp)
```

Comment:
this line seems wrong

Reply:
No problem, it is to prevent infinite loop.
`resp.Uncompressed` knows that there is a problem with the data requested for the first time
`r.Header.Get("Accept-Encoding") == ""` is to prevent an infinite loop and retry at most once
After &&, this if can only be satisfied once

Reply:
util.CloseResponse(resp) is to release the resources of the previous request

Reply:
Use `content-encoding` to upload a file for gzip, and then request without `accept-encoding`,can reproduce it

Reply:
what is the reason not to always set the "accept-encoding" to "gzip"?

Reply:
> what is the reason not to always set the "accept-encoding" to "gzip"?
There are too many buckets migrated from ceph, and there are many, many businesses. I am worried that a decompression that does not support `gzip`. They have not passed `accept-encoding` before, and now i cannot be pushed them to change. But like `cos`, `ceph`, etc., it is true that you can return directly without passing. The `content-length` of weed returns -1 and status  `404` because of the limitation of golang. The code in the content

Reply:
Ack. But this fix is not a clean way to solve this problem, just for backward compatibility.

Reply:
Yes, but I really didn't think of a more friendly solution. This change really helped us solve the problem of `ceph` and `cos`. If we can't merge into the `master`, we will have a little trouble using `seaweedfs` later.